cmd/ingress-operator/main.go

```diff
@@ -172,24 +183,30 @@ func createDNSProvider(cl client.Client, operatorConfig operatorconfig.Config, i
 	return dnsProvider, nil
 }
 
-// TODO: This can be replaced by cluster API when
-// https://github.com/openshift/installer/pull/1725 is available.
-type installConfig struct {
-	Platform struct {
-		AWS struct {
-			Region string `json:"region"`
-		} `json:"aws"`
-	} `json:"platform"`
-}
-
-func newInstallConfig(clusterConfig *corev1.ConfigMap) (*installConfig, error) {
-	data, ok := clusterConfig.Data["install-config"]
-	if !ok {
-		return nil, fmt.Errorf("missing install-config in configmap")
-	}
-	var ic installConfig
-	if err := yaml.Unmarshal([]byte(data), &ic); err != nil {
-		return nil, fmt.Errorf("invalid install-config: %v\njson:\n%s", err, data)
+// newProxyClient returns an http.Client based on http.DefaultClient with Proxy
+// set to a proxy function based on proxyConfig.
+func newProxyClient(proxyConfig *configv1.Proxy) *http.Client {
+	cfg := &httpproxy.Config{
+		HTTPProxy:  proxyConfig.Status.HTTPProxy,
+		HTTPSProxy: proxyConfig.Status.HTTPSProxy,
+		NoProxy:    proxyConfig.Status.NoProxy,
+	}
+	proxyFn := cfg.ProxyFunc()
+	proxy := func(req *http.Request) (*url.URL, error) {
+		return proxyFn(req.URL)
+	}
+	return &http.Client{
+		Transport: &http.Transport{
+			Proxy: proxy,
+			DialContext: (&net.Dialer{
+				Timeout:   30 * time.Second,
```

Comment:
Can you provide context for the `Timeout`, `KeepAlive` and `DualStack` values?

Reply:
The struct is copied from [DefaultTransport](https://golang.org/src/net/http/transport.go) in the stdlib. Not sure if there's any better way to get the defaults (unlike default http.Client which has a useful zero-value).

Reply:
@ironcladlou thanks for the reference. The clients are currently using the default http.Client, correct?